dhall/src/Dhall/Main.hs

```diff
@@ -210,6 +221,14 @@ parseOptions =
             <>  Options.Applicative.help description
             )
 
+    flag name description =
+        Options.Applicative.flag
+            (Just False)
+            Nothing
```

Comment:
I think the issue we're running into here is that we don't have a good way via the command line to specify the following three options:

* Use the default (possibly obtained from the configuration file)
* Override the default to prefer `False`
* Override the default to prefer `True`

… because if we use `flag` then the user can only indicate at most two of those options.

One way handle this is to add a new flag for the opposite of `--explain` and `--ascii`.  I'm thinking something like:

* `--unicode` (the opposite of `--ascii`)
* `--concise` (the opposite of `--explain`)

… and if the user doesn't specify either alternative then it uses whatever is in the default or the configuration file.

You can implement that behavior in `optparse-applicative` by doing something like this (for the case of ASCII vs. Unicode):

```haskell
    Options.Applicative.flag' (Just ASCII) …
<|> Options.Applicative.flag' (Just Unicode) …
<|> pure Nothing
```

Reply:
That's a nice catch for what would have been a UX bug!

In your (very helpful) code example, you used ASCII/Unicode instead of True/False. Currently, the options `ascii` constructor is `Bool`. I think it may actually be easier to keep it close to as-is (Maybe Bool) because then we can use the same `resolve` logic for both ascii and explain:

```hs
                -- resolve :: Maybe Bool -> (t -> Maybe Bool) -> Maybe t -> Bool
                resolve (Just option) _configKey _config = option
                resolve Nothing _configKey Nothing = False
                resolve Nothing configKey (Just config) = fromMaybe False $ configKey config
```

Reply:
@vmchale : Yeah, it's fine if you use `True`/`False` instead of `ASCII`/`Unicode`